tools/auto_split_onnx/auto_split_onnx.py

```diff
@@ -0,0 +1,296 @@
+        run_args.input_visibility = "private"
+        run_args.param_visibility = "private"
+        run_args.output_visibility = "private"
+        
+    elif subgraph_index != 1 & is_final_sub_model == False:
+        run_args.input_visibility = "hashed"
+        run_args.param_visibility = "private"
+        run_args.output_visibility = "hashed"
+        
+    elif subgraph_index != 1 & is_final_sub_model == True:
+        run_args.input_visibility = "hashed"
+        run_args.param_visibility = "private"
+        run_args.output_visibility = "public"
+        
+    else:
+        raise ValueError("Invalid subgraph index and final model flag.")
+    
+    res = ezkl.gen_settings(temp_model_name, py_run_args=run_args)
+    assert res == True
+    
+    data_path = f"input_data_{subgraph_index}.json"
+    generate_random_input_data(temp_model_name, data_path)
+    
+    res = ezkl.calibrate_settings(data_path, temp_model_name)
+    
+    setting_file_path = "settings.json"
+    
+    is_pass, pow = check_total_assignments(setting_file_path, upper_bound)
+    print(f"Subgraph {subgraph_index}: Inputs: {subgraph_inputs}, Outputs: {subgraph_outputs}, Pow: {pow}, Is Pass: {is_pass}")
+    
+    ## remove temp model and data
+    if not is_final_sub_model:
+        os.remove(temp_model_name)
+    # os.remove(data_path)
+    
+    return is_pass, pow
+
+def save_graph(G, file_name):
+    nx.draw(G, with_labels=True, font_size=5)
+    plt.savefig(file_name)
+    
+def print_node_names_and_types(onnx_model_path):
+    # Load the ONNX model
+    model = onnx.load(onnx_model_path)
+    graph = model.graph
+    
+    # Iterate through each node in the graph
+    for node in graph.node:
+        # Get the node's name and type
+        node_name = node.name if node.name else "Unnamed"
+        node_type = node.op_type
+        
+        # Print the node's name and type
+        print(f"Node Name: {node_name}, Node Type: {node_type}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Process ONNX model and generate subgraphs.")
+    parser.add_argument("--onnx_model_path", type=str, default='./resnet18.onnx', help="Path to the ONNX model. Default is './resnet18.onnx'.")
+    parser.add_argument("--simplified_model_path", type=str, default='./resnet18_simplified.onnx', help="Path to save the simplified ONNX model. Default is './resnet18_simplified.onnx'.")
+    parser.add_argument("--upper_bound_per_subgraph", type=int, default=23, help="Upper bound per subgraph in the form of 2^n. Default is 23.")
+    parser.add_argument("--input_shape", type=str, default='{"input": [1, 3, 224, 224]}', 
+                        help="Input shape for the ONNX model in JSON format. Default is '{\"input\": [1, 3, 224, 224]}'.")
+    parser.add_argument("--simplify", action='store_true', 
+                        help="Flag to indicate if the model should be simplified. Default is False.")
+    
```

Comment:
can we add an argument to specify the visibility of the stitched commitments eg. `public`, `hashed`, `polycommit`